locations/spiders/cava.py

```diff
@@ -0,0 +1,176 @@
+    "SC": "South Carolina",
+    "SD": "South Dakota",
+    "TN": "Tennessee",
+    "TX": "Texas",
+    "UT": "Utah",
+    "VA": "Virginia",
+    "VI": "Virgin Islands",
+    "VT": "Vermont",
+    "WA": "Washington",
+    "WI": "Wisconsin",
+    "WV": "West Virginia",
+    "WY": "Wyoming",
+}
+
+DAY_MAPPING = [
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday",
+]
+
+
+class CavaSpider(scrapy.Spider):
+    name = "cava"
+    allowed_domains = ["www.cava.com"]
+    item_attributes = {"brand": "Cava", "brand_wikidata": "Q85751038"}
+    start_urls = ("https://cava.com/locations/",)
+
+    def parse(self, response):
+        state_selectors = response.xpath(
+            './/div[@class="menu-panel-wrapper"]/div[@class="menu-content"]'
+        )
+        for state_selector in state_selectors:
+            yield from self.parse_state(state_selector)
+
+    def parse_state(self, state):
+        location_selectors = state.xpath('.//div[@class="vcard"]')
+        for location_selector in location_selectors:
+            yield from self.parse_location(location_selector)
+
+    def parse_location(self, location):
+        nonBreakingSpace = "\xa0"
+        name = location.xpath(".//h3/text()").extract_first()
+        street_address = (
+            location.xpath('.//div[@class="street-address"]/text()')
+            .extract_first()
+            .replace(nonBreakingSpace, " ")
+            + ", "
+            + location.xpath('.//span[@class="locality"]/text()')
+            .extract_first()
+            .replace(nonBreakingSpace, " ")
+        )
+        state = STATES.get(
+            location.xpath('.//span[@class="region"]/text()').extract_first().strip(),
+            "",
+        )
+        postcode = location.xpath(
+            './/span[@class="postal-code"]/text()'
+        ).extract_first()
+        addr_full = street_address + ", " + state + ", " + postcode
+        phone = location.xpath('.//a[contains(@href, "tel")]/@href').extract_first()
+        phone = phone.replace("tel:", "") if phone else ""
+        opening_hours = location.xpath('.//p[@class="copy"]/text()').extract_first()
+        if opening_hours:
+            if not "day" in opening_hours and not "Daily" in opening_hours:
+                opening_hours = opening_hours.replace("Hours:", "Hours: Daily, ")
+            opening_hours = opening_hours.replace("Hours: ", "").replace(
+                "Daily", "Monday - Sunday"
+            )
+            if "||" in opening_hours:
+                opening_hours = opening_hours.split("||")
+            else:
+                opening_hours = opening_hours.split("//")
+            opening_hours = self.parse_opening_hours(opening_hours)
+
+        properties = {
+            "ref": name,
+            "name": name,
+            "street_address": street_address,
+            "postcode": postcode,
+            "state": state,
+            "addr_full": addr_full,
```

Comment:
We don't need `addr_full` if we have `street_address`, `city`, and `state`.

Reply:
Sounds, good, I have removed it.